modules/041-linstor/templates/linstor-csi/linstorcsidriver.yaml

```diff
@@ -170,9 +170,27 @@ spec:
       maxUnavailable: 1
     {{- end }}
   controllerEndpoint: https://linstor.d8-{{ .Chart.Name }}.svc:3371
+  {{- if .Values.linstor.dataNodes }}
+  {{- if .Values.linstor.dataNodes.nodeSelector }}
+  nodeAffinity:
+    nodeAffinity:
+      requiredDuringSchedulingIgnoredDuringExecution:
```

Comment:
> IgnoredDuringExecution

This is a problem, because it requires the manual deletion of pods on node selector changes

Reply:
The thing you need is `requiredDuringSchedulingRequiredDuringExecution` which is does not exist :smile:

Reply:
It's not. On every pod template change DaemonSet controller generates a new controller revision, then reconciling pods to remove all the pods with old revision and create new pods with the new revision.

Reply:
In case when the node selector is changed yes, but if the label if gome from the node, the pod will stay.

Reply:
@nabokihms dunno, but unlabeling node works for me:

```yaml
apiVersion: apps/v1
kind: DaemonSet
metadata:
  name: test
  namespace: kube-system
spec:
  selector:
    matchLabels:
      name: test
  template:
    metadata:
      labels:
        name: test
    spec:
      tolerations:
      - key: node-role.kubernetes.io/control-plane
        operator: Exists
        effect: NoSchedule
      - key: node-role.kubernetes.io/master
        operator: Exists
        effect: NoSchedule
      containers:
      - name: test
        image: containous/whoami
      terminationGracePeriodSeconds: 0
      affinity:
        nodeAffinity:
          requiredDuringSchedulingIgnoredDuringExecution:
            nodeSelectorTerms:
              - matchExpressions:
                - key: foo
                  operator: In
                  values:
                  - bar
```

```console
# kubectl get ds -n kube-system
NAME      DESIRED   CURRENT   READY   UP-TO-DATE   AVAILABLE   NODE SELECTOR           AGE
test      0         0         0       0            0           <none>                  5h22m

# kubectl label node  virtlab-vo-2 foo=bar
node/virtlab-vo-2 labeled

# kubectl get ds -n kube-system
NAME      DESIRED   CURRENT   READY   UP-TO-DATE   AVAILABLE   NODE SELECTOR           AGE
test      1         1         1       1            1           <none>                  5h23m

# kubectl label node  virtlab-vo-1 foo=bar
node/virtlab-vo-1 labeled

# kubectl get ds -n kube-system
NAME      DESIRED   CURRENT   READY   UP-TO-DATE   AVAILABLE   NODE SELECTOR           AGE
test      2         2         2       2            2           <none>                  5h23m

# kubectl label node  virtlab-vo-2 foo-
node/virtlab-vo-2 unlabeled

# kubectl get ds -n kube-system
NAME      DESIRED   CURRENT   READY   UP-TO-DATE   AVAILABLE   NODE SELECTOR           AGE
test      1         1         1       1            1           <none>                  5h23m
```